### PR TITLE
Adds Google Analytics and Mixpanel tracking

### PR DESCRIPTION
**Pull request by oliverviljamaa**
# v0.9.0
## Adds Google Analytics and Mixpanel tracking

The optional tracking property is added to item objects and controlled there,
keeping the underlying `Navigation` component as flexible as possible for other nav use cases.

The tracking properties:

| Property    | Type      | Description                                      | Public Navigation                   |
|-------------|-----------|--------------------------------------------------|-------------------------------------|
| `category`  | `String`  | top level category                               | `'Public navigation'`               |
| `link`      | `String`  | link to be included in the event                 | any anchor href                     |
| `analytics` | `Boolean` | whether event should be sent to Google Analytics | `true` for every link               |
| `mixpanel`  | `Boolean` | whether event should be sent to Mixpanel         | `true` for `/login` and `/register` |

The events are published as:

| Google Analytics                   | Mixpanel                                       |
|------------------------------------|------------------------------------------------|
| `category`, `'click link'`, `link` | `` `${category} - clicked link` ``, `{ link }` |

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,24 @@
...
+The optional `tracking` property is added to item objects and controlled there,
+keeping the underlying `Navigation` component as flexible as possible for other nav use cases.
+
+The properties of `tracking`:
+
+| Property    | Type      | Description                                      | Public Navigation                   |
+|-------------|-----------|--------------------------------------------------|-------------------------------------|
+| `category`  | `String`  | top level category                               | `'Public navigation'`               |
+| `link`      | `String`  | link to be included in the event                 | any anchor href                     |
+| `analytics` | `Boolean` | whether event should be sent to Google Analytics | `true` for every link               |
```
Comment (OFFLlNE) [inline CHANGELOG.md]: I would rename analytics and mixpanel to more descriptive names like sendToAnalytics or sendAnalytics

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,24 @@
+# v0.9.0
+## Adds Google Analytics and Mixpanel tracking
+
+The optional `tracking` property is added to item objects and controlled there,
+keeping the underlying `Navigation` component as flexible as possible for other nav use cases.
+
+The properties of `tracking`:
+
+| Property    | Type      | Description                                      | Public Navigation                   |
```
Comment (Tankenstein) [inline CHANGELOG.md]: Nice table

File: src/Navigation/ButtonItem/ButtonItem.js [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
-const ButtonItem = ({ text, link, inverse }) => (
+import createTrackFunction from '../common/tracking';
+
+const ButtonItem = ({ text, link, inverse, tracking }) => (
   <div className="navbar-btn pull-xs-right">
-    <a href={link} className={`btn ${inverse ? 'btn-image' : 'btn-primary'}`}>
+    <a
+      href={link}
+      className={`btn ${inverse ? 'btn-image' : 'btn-primary'}`}
+      onClick={createTrackFunction(tracking)}
```
Comment (Tankenstein) [inline src/Navigation/ButtonItem/ButtonItem.js]: Should you call this if tracking is not passed?

File: src/Navigation/Menu/Items/Item/Dropdown/DropdownItem/DropdownItem.js [hunk 4]
```diff
@@ -1,15 +1,24 @@
...
 );
 
 DropdownItem.propTypes = {
   text: Types.string.isRequired,
   link: Types.string.isRequired,
+  tracking: Types.shape(),
+};
+
+DropdownItem.defaultProps = {
+  tracking: null,
```
Comment (Tankenstein) [inline src/Navigation/Menu/Items/Item/Dropdown/DropdownItem/DropdownItem.js]: Different default prop for this v button?

File: src/Navigation/Menu/Items/Item/Dropdown/DropdownItem/DropdownItem.js [hunk 5]
```diff
@@ -1,15 +1,24 @@
 import React from 'react';
 import Types from 'prop-types';
 
-const DropdownItem = ({ text, link }) => (
+import createTrackFunction from '../../../../../common/tracking';
+
+const DropdownItem = ({ text, link, tracking }) => (
   <li>
-    <a href={link}>{text}</a>
+    <a href={link} onClick={createTrackFunction(tracking)}>
```
Comment (Tankenstein) [inline src/Navigation/Menu/Items/Item/Dropdown/DropdownItem/DropdownItem.js]: Should you pass this if no tracking passed?

File: CHANGELOG.md (hunk 1 above)
Comment (oliverviljamaa) [reply CHANGELOG.md]: I'm not sure I agree, I like my JSON properties as non-verbs usually, but I see your concern.
@codewhisperer @Tankenstein @KarlEST WDYT?

File: CHANGELOG.md (hunk 2 above)
Comment (oliverviljamaa) [reply CHANGELOG.md]: Yes.

File: src/Navigation/ButtonItem/ButtonItem.js (hunk 3 above)
Comment (oliverviljamaa) [reply src/Navigation/ButtonItem/ButtonItem.js]: I had `tracking ? createTrackFunction(tracking) : () => {}` before, but it seemed too verbose and I thought it'd be fine as the function doesn't do anything when it gets an empty tracking object.

Will change if you think it's better, though, I'm super 50-50 on this.

File: src/Navigation/Menu/Items/Item/Dropdown/DropdownItem/DropdownItem.js (hunk 5 above)
Comment (oliverviljamaa) [reply src/Navigation/Menu/Items/Item/Dropdown/DropdownItem/DropdownItem.js]: Replied previously, pls respond.

File: src/Navigation/ButtonItem/ButtonItem.js (hunk 3 above)
Comment (Tankenstein) [reply src/Navigation/ButtonItem/ButtonItem.js]: fair.

File: src/Navigation/Menu/Items/Item/Dropdown/DropdownItem/DropdownItem.js (hunk 5 above)
Comment (Tankenstein) [reply src/Navigation/Menu/Items/Item/Dropdown/DropdownItem/DropdownItem.js]: fair

File: CHANGELOG.md (hunk 1 above)
Comment (Tankenstein) [reply CHANGELOG.md]: I actually don't like that it's called "analytics" and not "googleAnalytics", analytics is like a general term

Comment (Tankenstein) [reply CHANGELOG.md]: can see this being misleading

Comment (oliverviljamaa) [reply CHANGELOG.md]: So... `googleAnalytics` and `mixpanel` or `sendToGoogleAnalytics` and `sendToMixpanel`?

Comment (OFFLlNE) [reply CHANGELOG.md]: @Tankenstein

Comment (Tankenstein) [reply CHANGELOG.md]: sendToGoogleAnalytics + sendToMixpanel

**Comment by oliverviljamaa** [comment]
We may go with tracking on Google Tag Manager instead. Closing for now.

**Pull request closed without merging**